lib/Driver/ToolChains.cpp

```diff
@@ -1218,6 +1218,11 @@ std::string toolchains::GenericUnix::getDefaultLinker() const {
     // final executables, as such, unless specified, we default to gold
     // linker.
     return "gold";
+  case llvm::Triple::x86_64:
+  case llvm::Triple::ppc64:
+  case llvm::Triple::ppc64le:
+    // BFD linker has issues wrt relocations against protected symbols.
+    return "gold";
```

Comment:
Yes, its an ELF thing rather than a Linux thing, so I think that the current change as proposed makes sense.  However, Id rather leave the structure as is (I explicitly split up the switch as the reasoning is different).  That said, we could change the structure in a follow up change as well if others feel that is a good idea.